fastlane_core/lib/fastlane_core/helper.rb

```diff
@@ -39,6 +39,13 @@ def self.bundler?
       return false
     end
 
+    # Do we run from a bundled fastlane, which contains Ruby and OpenSSL?
+    # Usually this means the fastlane directory is ~/.fastlane/bin/
+    # We set this value via the environment variable `SELF_CONTAINED`
+    def self.contained_fastlane?
+      ENV.key?("SELF_CONTAINED")
```

Comment:
In general
```
ENV["SELF_CONTAINED"].to_s == "true
```
doesn't work when the user uses
```
export SELF_CONTAINED="1"
```
however for this case it's good enough